packages/x-tree-view/src/internals/utils/tree.ts

```diff
@@ -32,7 +32,27 @@ export const getPreviousNavigableItem = (
     return itemMeta.parentId;
   }
 
-  let currentItemId: string = siblings[itemIndex - 1];
+  // Finds the previous navigable sibling.
```

Comment:
If the previous sibling was disabled it was all broken

Reply:
Nice fix. 👍 
These are situations where you appreciate the effort put into writing tests. 👏 🎉